e2e/test/scenarios/joins/reproductions/23293-drill-through-implicit-join-add-remove-column.cy.spec.js

```diff
@@ -0,0 +1,96 @@
+import {
+  restore,
+  popover,
+  openOrdersTable,
+  visitDashboard,
+} from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { ORDERS, PRODUCTS } = SAMPLE_DATABASE;
+
+describe("issue 23293", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should retain the filter when drilling through the dashboard card with implicitly added column (metabase#23293)", () => {
+    openOrdersTable();
+
+    cy.findByTestId("viz-settings-button").click();
+    modifyColumn("Product ID", "remove");
+    modifyColumn("Category", "add");
+    cy.wait("@dataset");
+
+    saveQuestion().then(
+      ({
+        response: {
+          body: { id },
+        },
+      }) => {
+        const questionDetails = {
+          query: {
+            "source-table": `card__${id}`,
+            aggregation: [["count"]],
+            breakout: [
+              [
+                "field",
+                PRODUCTS.CATEGORY,
+                {
+                  "source-field": ORDERS.PRODUCT_ID,
+                },
+              ],
+            ],
+          },
+          display: "bar",
+        };
+
+        cy.createQuestionAndDashboard({ questionDetails }).then(
+          ({ body: { dashboard_id } }) => {
+            visitDashboard(dashboard_id);
+          },
+        );
+
+        cy.get(".bar").first().realClick();
+        popover()
+          .findByText(/^See these/)
```

Comment:
nit: I think we should make it more obvious what we're clicking, so if I remember correctly, if you name the saved question you can use the pluralized name here. e.g. if the first question name is `Order question`, we could look for `"See these Order questions"`. I'm not suggesting we use that name, but that's just an example and I didn't use the name `Order` since that could be confused with the database table, for instance.